testing/testcafe/tests/dataGrid/editing.ts

```diff
@@ -1389,3 +1389,37 @@ test('Async Validation(Batch) - Validation frame should be rendered when a neigh
     }],
   }],
 })));
+
+// T905677
+test('Rollback changes on a click on a revert button  when startEditAction is dblclick', async (t) => {
+  const dataGrid = new DataGrid('#container');
+  const dataRow = dataGrid.getDataRow(0);
+  const cell0 = dataRow.getDataCell(1);
+  const $revertButton = Selector('.dx-revert-button');
+
+  await t
+    .doubleClick(cell0.element)
+    .expect(cell0.isEditCell).ok()
+    .click(cell0.element.find('.dx-checkbox'))
+    .expect($revertButton.exists)
+    .ok()
+    .click($revertButton)
+    .expect($revertButton.exists)
+    .notOk()
+    .expect(cell0.isEditCell)
+    .notOk();
+}).before(() => createWidget('dxDataGrid', {
```

Comment:
I would add checking the current cell value using the following code:
```suggestion
    .notOk()
    .expect(dataGrid.apiGetCellValue(0, 1))
    .notOk(); //means that the cell value is undefined after clicking the revert button
}).before(() => createWidget('dxDataGrid', {
```